bundles/org.openhab.core.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java

```diff
@@ -35,14 +40,89 @@
  * except if a {@link ConfigDescriptionParameter} is given. In the latter case,
  * a conversion according to the type given in the description is performed.
  *
- * @author Kai Kreuzer - Initial API and implementation
+ * @author Kai Kreuzer - Initial contribution
  * @author Thomas Höfer - Minor changes for type normalization based on config description
  */
 public class ConfigUtil {
+
+    private static final String DEFAULT_LIST_DELIMITER = ",";
+
+    /**
+     * Maps the provided (default) value of the given {@link Type} to the corresponding Java type.
+     *
+     * In case the provided value is supposed to be a number and cannot be converted into the target type correctly,
+     * this method will return <code>null</code> while logging a warning.
+     *
+     * @param parameterType the {@link Type} of the value (must not be null)
+     * @param defaultValue the value that should be converted (must not be null)
+     * @return the given value as the corresponding Java type or <code>null</code> if the value could not be converted
+     */
+    public static @Nullable Object normalizeDefaultType(Type parameterType, String defaultValue) {
+        try {
+            switch (parameterType) {
+                case TEXT:
+                    return defaultValue;
+                case BOOLEAN:
+                    return Boolean.parseBoolean(defaultValue);
+                case INTEGER:
+                    return new BigInteger(defaultValue);
+                case DECIMAL:
+                    return new BigDecimal(defaultValue);
+                default:
+                    return null;
+            }
+        } catch (NumberFormatException e) {
+            LoggerFactory.getLogger(ConfigUtil.class).warn("Could not parse default value '{}' as type '{}': {}",
+                    defaultValue, parameterType, e.getMessage(), e);
+            return null;
+        }
+    }
+
+    /**
+     * Applies the default values from a give {@link ConfigDescription} to the given {@link Configuration}.
+     *
+     * @param configuration the {@link Configuration} where the default values should be added (must not be null)
+     * @param configDescription the {@link ConfigDescription} where the default values are located (may be null, but
+     *            method won't have any effect then)
+     */
+    public static void applyDefaultConfiguration(Configuration configuration,
```

Comment:
`limitToOptions`: Hm, hopefully there are no default values that aren't in the option list... I think requesting the (possibly dynamic) options might be too diffcult/expensive here, so I would not do this check and be optimistic.

`multipleLimit`: This would be easier to perform, but we have no clue if we might remove an important one and should have taken a different one. So again, I would just assume that the defaults are correctly defined and not do anything further here.

Reply:
Yes, `limitToOptions` will be a quite a challenge. And we have the same issues with dynamic config parameters. AFAIK they will not validated too.

Regarding `multipleLimit` we maybe can log a warning instead of stripping them.

Reply:
Yes, logging a warning isn't a bad idea.

Reply:
Done.